dev-client/src/screens/AddUserToProjectScreen/AddUserToProjectRoleScreen.tsx

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright © 2024 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {useCallback, useState} from 'react';
+import {useTranslation} from 'react-i18next';
+
+import {Button} from 'native-base';
+
+import {
+  addUserToProject,
+  ProjectRole,
+} from 'terraso-client-shared/project/projectSlice';
+
+import {ScreenContentSection} from 'terraso-mobile-client/components/content/ScreenContentSection';
+import {
+  Box,
+  Column,
+  Row,
+} from 'terraso-mobile-client/components/NativeBaseAdapters';
+import {AppBar} from 'terraso-mobile-client/navigation/components/AppBar';
+import {useNavigation} from 'terraso-mobile-client/navigation/hooks/useNavigation';
+import {MinimalUserDisplay} from 'terraso-mobile-client/screens/AddUserToProjectScreen/components/MinimalUserDisplay';
+import {ProjectRoleRadioBlock} from 'terraso-mobile-client/screens/AddUserToProjectScreen/components/ProjectRoleRadioBlock';
+import {UserFields} from 'terraso-mobile-client/screens/AddUserToProjectScreen/components/UserDisplay';
+import {ScreenScaffold} from 'terraso-mobile-client/screens/ScreenScaffold';
+import {useDispatch, useSelector} from 'terraso-mobile-client/store';
+
+type Props = {
+  projectId: string;
+  user: UserFields;
+};
+
+export const AddUserToProjectRoleScreen = ({projectId, user}: Props) => {
+  const {t} = useTranslation();
+  const dispatch = useDispatch();
+  const navigation = useNavigation();
+
+  const project = useSelector(state => state.project.projects[projectId]);
+
+  const [selectedRole, setSelectedRole] = useState<ProjectRole>('VIEWER');
+
+  const onCancel = useCallback(() => {
+    navigation.pop();
+  }, [navigation]);
+
+  const addUser = useCallback(async () => {
+    try {
+      dispatch(
+        addUserToProject({userId: user.id, role: selectedRole, projectId}),
+      );
+    } catch (e) {
+      console.error(e);
+    }
+    navigation.pop();
+    navigation.pop();
```

Comment:
question: does it make sense to `navigate()` to a.specific page instead of `pop()`-ing twice? That's how we handle the site creation flow.